assets/src/edit-story/components/inspector/prepublish/checklistTab.js

```diff
@@ -194,14 +210,63 @@ const ChecklistTab = (props) => {
     [checklist]
   );
 
+  const selectElement = useCallback(
+    ({ elementId, elements, pageId }) => {
+      if (pageId) {
+        setCurrentPage({ pageId });
+      }
+      if (Array.isArray(elements)) {
+        setSelectedElementsById({
+          elementIds: elements.map(({ id }) => id),
+        });
+      } else if (elementId) {
+        setSelectedElementsById({ elementIds: [elementId] });
+      }
+    },
+    [setCurrentPage, setSelectedElementsById]
+  );
+
+  const getOnClick = useCallback(
+    ({ elementId, pageId, elements }) => {
+      if (!elementId && !pageId && !elements) {
+        return undefined;
+      }
+      return () => selectElement({ elementId, pageId, elements });
+    },
+    [selectElement]
+  );
+
+  const getHandleKeyPress = useCallback(
+    ({ elementId, elements, pageId }) => (event) => {
+      if (event.key === 'Enter') {
+        selectElement({ elementId, elements, pageId });
+      }
+    },
+    [selectElement]
+  );
+
   const renderRow = useCallback(
-    ({ message, help, id, pageGroup }) => (
-      <Row key={`guidance-${id}`} pageGroup={pageGroup}>
-        {message}
-        <HelperText>{help}</HelperText>
-      </Row>
-    ),
-    []
+    ({ message, help, id, pageGroup, ...args }) => {
+      const onClick = getOnClick(args);
+      const handleKeyPress = getHandleKeyPress(args);
+      return (
+        <Row
+          tabIndex={0}
+          onClick={onClick}
+          // note: onKeyDown does not work for "Enter"
+          onKeyUp={handleKeyPress}
+          aria-label={
+            onClick ? __('Select element for error', 'web-stories') : undefined
```

Comment:
I'm not a native English speaker, but this sounds odd. Maybe `Select offending element`?

However, doesn't the use of `aria-label` here mean that `message` and `help` will never be announced? If so, that sounds bad from an a11y perspective.

Reply:
hm, good point. I'll look into this.

Reply:
I'm going to move the aria-label to the helper text span, which should work

Reply:
@embarks Won't that override the helper text still? How is it announced by VoiceOver?

Reply:
Since I don't have requirements for this behavior, I'm making my own assumptions on how it should work, but here's how it looks using tab + arrow keys:
![voiceovertest](https://user-images.githubusercontent.com/41136059/104249448-a6df2900-5428-11eb-85de-6a5082b6be92.gif)
One tabs through the list, then when the voice over reads "group" that indicates you can select the next item in the group with arrow keys, in this case arrow key "right". The way it is implemented now it reads it in the following order:

`tab`
> Voiceover reads item title, "group"
`arrow key right`
> Voiceover reads aria-label, "group"
`arrow key right`
>Voice over reads text, "clickable"

Then, pressing enter selects the item.

Reply:
So I think what we need here is not `aria-label`, but text that is visually hidden, like `VisuallyHiddenDescription` from the dashboard.

Then we'd do:

```js
const VisuallyHidden = styled.span`
  position: absolute;
  height: 1px;
  width: 1px;
  overflow: hidden;
  clip: rect(1px, 1px, 1px, 1px);
  white-space: nowrap;
`;

<Row
  tabIndex={0}
  onClick={onClick}
  onKeyUp={handleKeyPress}
  key={id}
  pageGroup={pageGroup}
>
  {message}
  <HelperText>{help}</HelperText>
  {onClick && <VisuallyHidden>{__('Select offending element', 'web-stories')}</VisuallyHidden>}
</Row>
```

This is the only way to get all three texts announced.

Reply:
works great :+1: thank you